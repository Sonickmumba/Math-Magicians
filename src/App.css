body {
  margin: 0;
  padding: 0;
}

*::before,
*::after {
  box-sizing: border-box;
}

.calculator-grid-container {
  display: grid;
  width: 300px;
  grid-template-rows: repeat(6, 1fr);
  grid-template-columns: repeat(auto-fit, 75px);
  gap: 0;
  margin: 40px auto;
}

.operand {
  width: 100%;
  background-color: rgb(133, 134, 146);
  outline: none;
  border: 0;
  text-align: right;
  font-size: 1.5rem;
}

.btn {
  margin: 0;
  border-radius: 0;
  height: 50px;
  background-color: rgb(224, 224, 224);
  border: 1px solid rgb(202, 202, 202);
}

.operation {
  margin: 0;
  border-radius: 0;
  height: 50px;
  background-color: rgb(244, 141, 75);
  border: 1px solid rgb(202, 202, 202);
}

.output-result {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  grid-column: span 4;
  text-align: right;
  height: 50px;
  font-size: 1.5rem;
  background-color: rgb(133, 134, 146);
}

.btn-span-two {
  grid-column: span 2;
  margin: 0;
  background-color: rgb(224, 224, 224);
  border: 1px solid rgb(202, 202, 202);
}
