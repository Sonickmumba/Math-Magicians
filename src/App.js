import React from 'react';
import './App.css';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Calculator from './components/Calculator';
import Header from './pages/Header';
import Home from './pages/Home';
import Quote from './pages/Quote';

const App = () => (
  <BrowserRouter>
    <Header />
    <Routes>
      <Route path="/" element={<Calculator />} />
      <Route path="/home" element={<Home />} />
      <Route path="/quote" element={<Quote />} />
    </Routes>
  </BrowserRouter>
);

export default App;
