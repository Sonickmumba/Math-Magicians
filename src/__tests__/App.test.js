import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import App from '../App';

test('Test for App Component', () => {
  render(<App />);
  const linkElement = screen.getByText(/Welcome To Our Page/i);
  expect(linkElement).toBeInTheDocument();
});

describe('Test for  App component', () => {
  it('App Component renders correctly', () => {
    const tree = TestRenderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
