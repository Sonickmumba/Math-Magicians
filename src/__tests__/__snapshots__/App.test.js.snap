// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for  App component App Component renders correctly 1`] = `
Array [
  <header
    className="header-container"
  >
    <h1
      className="header-title"
    >
      Math Magicians
    </h1>
    <nav
      id="navbar-container"
    >
      <ul
        className="navbar-items"
      >
        <li
          className="nav-link"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <span
          className="vertical-line"
        />
        <li
          className="nav-link"
        >
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <span
          className="vertical-line"
        />
        <li
          className="nav-link"
        >
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    id="home-link-container"
  >
    <h2
      className="home-title"
    >
      Welcome To Our Page
    </h2>
    <p
      className="home-link-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <p
      className="home-link-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </main>,
]
`;

exports[`Unit Test For  App component App Component renders correctly 1`] = `
Array [
  <header
    className="header-container"
  >
    <h1
      className="header-title"
    >
      Math Magicians
    </h1>
    <nav
      id="navbar-container"
    >
      <ul
        className="navbar-items"
      >
        <li
          className="nav-link"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <span
          className="vertical-line"
        />
        <li
          className="nav-link"
        >
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <span
          className="vertical-line"
        />
        <li
          className="nav-link"
        >
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    id="home-link-container"
  >
    <h2
      className="home-title"
    >
      Welcome To Our Page
    </h2>
    <p
      className="home-link-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <p
      className="home-link-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </main>,
]
`;
